src/chat/_chatwidget.py

```diff
@@ -231,6 +233,35 @@ def join(self, channel):
     def log_event(self, e):
         self.serverLogArea.appendPlainText("[%s: %s->%s]" % (e.eventtype(), e.source(), e.target()) + "\n".join(e.arguments()))
 
+    def updateAutoJoinIRCMenu(self):
+        self.client.menuAutoJoinIRC.clear()
+
+        # add channels we are currently in
+        for channel in self.channels:
```

Comment:
I think it would be more reasonable to either make this list static rather than generated dynamically, or let the user specify a list of names directly in some way (e.g. comma-separated).

Reply:
As you prefer, to be honest I liked this dynamic one (because very simple to use except a few click) but that's fine, will see how I can edit it to make it not dynamic like this. Probably something in the menu where when you click, it shows a popup or something asking to enter the channels you want to autojoin, and then you will see the list of these channels, but then need the possibility to remove them too.. mmh.

Reply:
Well, we can always just pre-populate the popup with existing autojoin channels. I'm not too fond of dynamic menu, especially dependent on current channels - it makes configuration confusingly dependent on what channels we're in, and prevents us from changing settings of channels we left unless we rejoin them.

Reply:
I see, I did another solution instead of this dynamic one. Basically you click on set autojoin channels then popup, and you can write what you want to join, and you can also see the list.